src/schemas/general/Frequency.ts

```diff
@@ -1,8 +1,16 @@
 import { z } from 'zod';
 
-const ONE_HOUR = 60 * 60;
-export const MAX_BASE_FREQUENCY = ONE_HOUR;
+import { formatDuration } from 'utils';
+import { ONE_HOUR_IN_MS, ONE_SECOND_IN_MS } from 'utils.constants';
 
-export const frequencySchema = z
-  .number()
-  .max(MAX_BASE_FREQUENCY, { message: `Frequency cannot be greater than ${MAX_BASE_FREQUENCY} seconds` });
+export const MIN_BASE_FREQUENCY = ONE_SECOND_IN_MS * 10;
+export const MAX_BASE_FREQUENCY = ONE_HOUR_IN_MS;
+
+export const MIN_FREQUENCY_ERROR_MESSAGE_START = `Frequency must be at least `;
+export const MAX_FREQUENCY_ERROR_MESSAGE_START = `Frequency cannot be greater than `;
+
+export const frequencySchema = (minFrequency = MIN_BASE_FREQUENCY) =>
```

Comment:
This is not a schema. It is a schema creator.
IMO the name should be something like `createFrequencySchema`